src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -54,7 +56,13 @@
     private final JsonFactory jsonFactory;
     private final ObjectReader eventReader;
 
+    private final Optional<MetricsCollector> metricsCollector;
```

Comment:
Instead of the `Optional` we could have an empty implementation of the collector, that way we can get rid of all the `ifPresent` checks.